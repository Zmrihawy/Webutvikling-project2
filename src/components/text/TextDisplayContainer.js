import React, { Component } from 'react';
import texts from '../../media/txt/texts.json';

import TextDisplay from './TextDisplay';

class TextDisplayContainer extends Component {

  constructor(props) {
    super(props);
    this.state = {
      randomText: ""
    }
  }

  componentDidMount() {
    const { textState } = this.props;  
    let keys = Object.keys(textState).filter(key => (
      textState[key]));
    if (keys.length == 1 && keys[0] == "random") {
      fetch("https://corporatebs-generator.sameerkumar.website/")
        .then(res => res.json())
        .then(res => (this.setState({randomText: res.phrase})))
    }
  }

  componentDidUpdate(prevProps) {
    const { textState, combinationState } = this.props;  
    if (prevProps.textState !== textState || prevProps.combinationState !== combinationState) {
      let keys = Object.keys(textState).filter(key => (
        textState[key]));
      if (keys.length == 1 && keys[0] == "random") {
        fetch("https://corporatebs-generator.sameerkumar.website/")
          .then(res => res.json())
          .then(res => (this.setState({randomText: res.phrase})))
      }
    }
  }

  render() {
    const { textState, combinationState } = this.props;  
    const { randomText } = this.state;
    let text = "";
    let keys = Object.keys(textState).filter(key => (
      textState[key]));
    if (keys.length > 1) {
      console.log("Critical error! Active text is more than one. This should never happen.")
    }

    if (keys.length != 0 && combinationState != null) {
      if (keys[0] == "random") {
        text = randomText; 
      } else {
        text = texts[keys[0]][combinationState];
      }
    }

    return (
    <div>
      <TextDisplay text={text}/>
    </div>
    )
  }
}

export default TextDisplayContainer;
