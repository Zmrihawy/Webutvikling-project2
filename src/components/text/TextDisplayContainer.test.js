import React from 'react';
import TextDisplayContainer from './TextDisplayContainer';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    let text = {
        boring: false,
        funny: true,
        random: false
    }
  const tree = renderer
    .create(<TextDisplayContainer textState={text} combinationState={1}></TextDisplayContainer>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});