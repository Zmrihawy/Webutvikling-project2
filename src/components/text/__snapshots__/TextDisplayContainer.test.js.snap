// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="text"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    listen, Jerry, I don't want to overstep my bounds or anything. It's yout hous. It's your world. You're a rea Julius Caesar, but I'll tell you some, tell you how-how I feel about school, Jerry. It's a waste of time, a bunch of people running around, bumping into each other. G-guy up front says, "two plus two". The people in the back say "four". Then the bell rings, and they give you a carton of milk and a piece of paper that says you can take a dump or something. I mean, it's-it's not place for smart people, Jerry, and I know that's not a popular opinion, but it's my two cents on the issue.
  </div>
</div>
`;
